dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFilePersistenceStrategy.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.file.batch
+
+import com.datadog.android.core.internal.domain.PayloadDecoration
+import com.datadog.android.core.internal.persistence.DataReader
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.persistence.PersistenceStrategy
+import com.datadog.android.core.internal.persistence.Serializer
+import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+import com.datadog.android.log.Logger
+
+internal open class BatchFilePersistenceStrategy<T : Any>(
```

Comment:
I guess you still have in mind to use the `DataProcessor` or to somehow inject the `ExecutorService` here. Right those `writers/readers` are working on the main thread.

Reply:
Those writer and readers are not working on any specific threads, just on the one they're being called on. But yes, I'll make sure they're called on the proper `executorService`.